aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -713,5 +713,58 @@ Tensor softmax_nested(const Tensor& input, const int64_t dim, const bool half_to
   return wrap_buffer(output_buffer, sizemat.clone());
 }
 
+Tensor bmm_nested(const Tensor& self, const Tensor& mat2) {
+  auto self_ptr = get_nested_tensor_impl(self),
+      mat2_ptr = get_nested_tensor_impl(mat2);
+  TORCH_CHECK(self_ptr->dim() == 3, "batch1 must be a 3D tensor");
+  TORCH_CHECK(mat2_ptr->dim() == 3, "batch2 must be a 3D tensor");
+  int64_t ntensors = self_ptr->size(0),
+      ntensors2 = mat2_ptr->size(0);
+  TORCH_CHECK(ntensors == ntensors2,
+      "Expected size for the 1st dimension of batch2 tensor to be: ", ntensors,
+      " but got: ", ntensors2, ".");
+  std::vector<int64_t> self_offsets = NestedTensor_get_offsets(self_ptr),
+      mat2_offsets = NestedTensor_get_offsets(mat2_ptr);
+  std::vector<IntArrayRef> self_shapes = NestedTensor_get_shapes(self_ptr),
+      mat2_shapes = NestedTensor_get_shapes(mat2_ptr);
+  const Tensor& self_buffer = self_ptr->get_buffer(),
+      & mat2_buffer = mat2_ptr->get_buffer();
+  // determine output size
+  const Tensor& self_sizemat = self_ptr->get_nested_size_tensor();
+  Tensor out_sizemat = self_sizemat.new_empty(self_sizemat.sizes());
+  int64_t* out_sizemat_ptr = out_sizemat.data_ptr<int64_t>();
+  std::vector<int64_t> out_offsets(ntensors + 1);
+  std::vector<IntArrayRef> out_shapes(ntensors);
+  out_offsets[0] = 0;
```

Comment:
I don't think this needs to be a std_vector. You're only using the `i-1`th value once to calculate the `i-th` value and then only use that last value to determine the size of `out_buffer`. You could just create a `int64_t out_numel` into which you accumulate the sizes of the output.

Reply:
`out_offsets` will be used again in line 761 to determine which segment of `out_buffer` catches the `i`th matmul result